src/docs/patterns/loading.mdx

```diff
@@ -0,0 +1,123 @@
+
+There are a number of libraries available to help you create this placeholder content, such as:
+
+-   [@netbasal/ngx-content-loader](https://www.npmjs.com/package/@netbasal/ngx-content-loader) (Angular, Ionic)
+-   [ion-skeleton-text](https://ionicframework.com/docs/api/skeleton-text) (Ionic)
+-   [placeholder-loading](https://www.npmjs.com/package/placeholder-loading) (React, Angular, Ionic)
+-   [react-content-loader](https://www.npmjs.com/package/react-content-loader) (React)
+-   [rn-placeholder](https://www.npmjs.com/package/rn-placeholder) (React Native)
+
+<React.Fragment>
+    <ImageGrid
+        images={[Image1, Image2]}
+        caption={'When data is loading, use placeholder shapes for where the data will appear.'}
+    />
+    <DemoCard
+        repository={'loading-states'}
+        angular={
+            'https://stackblitz.com/github/pxblue/angular-design-patterns?file=src%2Fapp%2Fpages%2Floading-states%2Floading-states.component.ts'
+        }
+        react={
+            'https://codesandbox.io/s/github/pxblue/react-design-patterns/tree/master?file=/src/pages/loading-states/index.tsx&initialpath=loading-states'
+        }
+        ionic
+        reactNative
+    />
+</React.Fragment>
+
+### Full Screen
+
+Full screen progress indicators can be used when transitioning between multiple screens in a workflow (such as user registration) or refreshing / reloading a particular element on a page. These make use of a semi-transparent overlay with a large, centered spinner. This variation of indicator is useful when you want to block user interaction with the screen (or a section of the screen) while processing occurs.
+
+TODO: Image?
+
+### Inline Spinners
+
+Inline spinners can be used as an alternative to full-screen loading indicators. The most common application of inline spinners is during [Form](/patterns/forms) submission / verification. If you need to wait for a server response to validate inputs in a form (after clicking the submit button), you should disable the submit button and replace its label with a spinner. If the verification succeeds, the application should proceed to the next screen. If verification fails, the appropriate error messages should be displayed and the submit button label restored.
+
+<ImageGrid
+    images={[InlineSpinner]}
+    regularWidth
+    caption={[
+        'After the user clicks the submit button, the app waits for the server response, and displays an error message.',
+        '1: The helper text will be replaced by the error message. ',
+        '2: Use a circular progress indicator when waiting for responses from the server.',
+    ]}
+/>
+
+### Progress Bars
+
+Progress bars should be used sparingly in your application. They should only be used in situations where progress is discrete / measurable. For example, it could be used to indicate progress through a multi-step workflow or when uploading a file to a server.
+
+TODO: Image?
+
+<Divider />
+
+# Behavior
```

Comment:
It feels like there is a mix of talk about the appearance and behaviors in the anatomy section above

Reply:
It's difficult to separate it cleanly because of the different variants. They are different in both how they look and how they behave. So we either discuss both in once area, or split all of the variants and talk about them in multiple places (which feels weird to me). Open to suggestions.

Reply:
I can go with both in one area.

Reply:
Maybe if we have "Variations", that's a higher level hierarchy than Anatomy/Behavior...so it could be something like

- Variation 1
    - Anatomy
    - Behavior
- Variation 2
    - Anatomy
    - Behavior

(we don't necessarily have to explicitly call out the Anatomy and Behavior sections if we don't want to)

Reply:
Can do